### PR TITLE
feat: adapt image example to Flow v1 (and use redis)

### PR DESCRIPTION
**Pull request by cristianmtr**
Note: I can only get this work/pass on https://github.com/jina-ai/jina/pull/1722. I am working off of that branch

Also requires https://github.com/jina-ai/jina-hub/pull/2582 for Redis

### CODE REVIEW

**Review by JoanFM — CHANGES_REQUESTED** [review]
use on_done instead of output_fn

File: stress-example/image_search/app.py [hunk 1]
```diff
@@ -61,31 +29,33 @@ def config(indexer_query_type):
...
-
-docs_index = random_docs(0, NUM_DOCS)
-docs_query = random_docs(0, QUERY_NUM_DOCS)
+    elif task == 'query':
+        raise ValueError(f'Indexer type {indexer_query_type} not supported')
 
 
 # for index
 def index():
     flow_index = Flow.load_config('flows/index.yml')
```
Comment (JoanFM) [inline stress-example/image_search/app.py:38]: can u set ? I think this should work
```python
Flow.load_config('flows/index.yml', port_expose=os.environ['JINA_PORT'])
```

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by JoanFM — COMMENTED** [review]
Can u add a small script to have an idea on what needs to be set on Redis side?

File: stress-example/image_search/app.py (hunk 1 above)
Comment (cristianmtr) [reply stress-example/image_search/app.py:41]: It wasn't set in the `index.yml` Flow. I added it and now it uses that

**Comment by cristianmtr** [comment]
> Can u add a small script to have an idea on what needs to be set on Redis side?

Added. It's basically the same as in the Hub image. But this won't be automated, right? It will be set up once on the respective AWS instance?

**Comment by JoanFM** [comment]
> > Can u add a small script to have an idea on what needs to be set on Redis side?
> 
> Added. It's basically the same as in the Hub image. But this won't be automated, right? It will be set up once on the respective AWS instance?

not sure, I just mean to have the knowledge in one place

File: stress-example/image_search/client.py [hunk 2]
```diff
@@ -34,40 +38,48 @@ def random_docs(start, end):
...
 @click.option('--port', '-p')
-def main(task, port):
-    if task == 'index':
-        index(port)
-    elif task == 'query':
-        query(port)
-    else:
+@click.option('--load', '-l', default=60)  # time (seconds)
+@click.option('--nr', '-n', default=DEFAULT_NUM_DOCS)
+def main(task, port, load, nr):
```
Comment (JoanFM) [inline stress-example/image_search/client.py:54]: do we want to set the parallelization at this level?

Comment (cristianmtr) [reply stress-example/image_search/client.py:54]: yes will add it now. Committed this as the refactoring required to do that

**Review by JoanFM — APPROVED** [review]

File: stress-example/image_search/flows/query.yml [hunk 3]
```diff
@@ -22,6 +33,7 @@ pods:
     uses: pods/doc.yml
     shards: $JINA_SHARDS_INDEXERS
     separated_workspace: true
+>>>>>>> 863c6075f60b6040dc7c639e02321b0c4b97afcb
```
Comment (JoanFM) [inline stress-example/image_search/flows/query.yml:36]: wrong merge from my side

File: stress-example/image_search/pods/merge_and_topk.yml [hunk 4]
```diff
@@ -12,6 +12,7 @@ requests:
         with:
           field: 'score__value'
           traversal_paths: ['m']
+          reverse: true
```
Comment (JoanFM) [inline stress-example/image_search/pods/merge_and_topk.yml:15]: Depending on metric this should be True or False

**Review by JoanFM — APPROVED** [review]

**Pull request merged**